### ISSUE TITLE
EditorNavigationContribution.onStart is slow

### ISSUE THREAD

**Issue by akosyakov** [theia-ide/theia#3413]
It takes over 1s each time.

**Comment by kittaakos** [comment]
As we have exceeded the 8 seconds, I looked into it. It seems it is not the editor navigator contributions which is slow but waiting for the workspace root to be set for the workspace storage service. For instance, if I remove the `await` from here:

https://github.com/theia-ide/theia/blob/e147cf39c7f484374b7ec0cd0d04958c3ef00bca/packages/editor/src/browser/editor-navigation-contribution.ts#L79

Then I can see the `bundle.js:95521 root WARN DebugFrontendApplicationContribution.onStart is slow, took: 8154.099999985192 ms` warning with the same ~8 seconds.

**Comment by akosyakov** [comment]
Can we restore navigation history async without blocking `onStart`?

**Comment by kittaakos** [comment]
Yes, we can but it won't change much. The overall app startup time will remain the same. If not the `EditorNavigationContribution` blocks, then something else waiting for the WS root will.

According to my measurements (console logging), the bottleneck is here:
https://github.com/theia-ide/theia/blob/e147cf39c7f484374b7ec0cd0d04958c3ef00bca/packages/workspace/src/browser/workspace-service.ts#L142

**Comment by sgraband** [comment]
Fixed in commit 990908be18; `onStart` is now non-blocking.
